### PR TITLE
[ESW-581] ECG Bit Reduction Standalone Algorithm

### PR DESCRIPTION
**Pull request by rhirani-myant**
# Summary
Adds a standalone application to test ECG Bit Reduction algorithm.

# Added
- ECG bit reduction source and header file
- Standalone application (main.c + Makefile) to run ECG bit reduction algorithm 

# Changed
- Move CSV Writer and Data Processing files to shared folder as they are used across multiple algorithms 
- Adjust Makefile + Includes for ABR

# Removed
- N/A

# Testing
- Built locally
- @glor-w to validate behavior and update script based on needs of Data Science team

### CODE REVIEW

File: ecg_bit_reduction/main.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    // Read Inputs from CSV:
+    printf("Setting inputs...\r\n");
+    CSVW_ReadCSV(INPUT_FILE_NAME, dpInput, NULL, NULL, &bNumRows);
+
+    // Loop through entire input array
+    printf("Looping through input data...\r\n\r\n");
+    for (uint32_t i = 0; i < bNumRows; i++)
+    {
+        // ECG Bit Reduction Algorithm
+        ecg_reduced_data = ecgbr_sample_reduction((uint32_t)dpInput[i], ECG1, ecg_buffer_refresh);
```
Comment (rhirani-myant) [inline ecg_bit_reduction/main.c:33]: Not sure what the point of `ecg_buffer_refresh` is. Looking into it

Comment (rhirani-myant) [reply ecg_bit_reduction/main.c:33]: Resolved; `ecg_buffer_refresh` is only true for the first iteration on boot, set to false afterwards

File: ecg_bit_reduction/ecg_bit_reduction.c [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+// --- Globals ---
+
+static const double a_hp[] = {1.0l,                 -1.9991669594972l,  0.9991673063310l};
+static const double b_hp[] = {0.99958356645707l,    -1.99916713291414l, 0.99958356645705l};
+static double hp_in[MAX_ECG][HP_FILTER_SIZE] = {0};
+static double hp_out[MAX_ECG][HP_FILTER_SIZE] = {0};
+
+// --- Functions ---
+
+static bool check_restart_filter( uint32_t raw_sample, ecg_sens_id ecg_id, bool restart)
```
Comment (toastwmy) [inline ecg_bit_reduction/ecg_bit_reduction.c:47]: Naming convention again (variable names)

File: ecg_bit_reduction/ecg_bit_reduction.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+{
+    //mean sample diff
+    static double mean[MAX_ECG] = {0};
+    static uint32_t ecg_prev[MAX_ECG] = {0};
+    static bool last_highamp_flag[MAX_ECG] = {0};
+    static uint8_t sample_count[MAX_ECG] = {0};
```
Comment (toastwmy) [inline ecg_bit_reduction/ecg_bit_reduction.c:53]: Declaring static arrays in a function isn't bad as long as they're const length; however, we've run into memleaks before when a static array was declared with a variable - perhaps move them out and into a context struct.

File: ecg_bit_reduction/ecg_bit_reduction.c (hunk 2 above)
Comment (toastwmy) [reply ecg_bit_reduction/ecg_bit_reduction.c:47]: This repeats several times below as well.

**Review by toastwmy — CHANGES_REQUESTED** [review]
Functionally looks good, will approve and merge when code style is changed.

**Review by toastwmy — APPROVED** [review]
LGTM!

**Pull request merged**